File: 0009-palindrome-number/0009-palindrome-number.go
func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }
    
    if x < 10 {
        return true
    }

    if x % 10 == 0 {
        return false
    }

    revHalf := 0

    for x > revHalf {
        revHalf = revHalf * 10 + (x % 10)
        x /= 10
    }

    return x == revHalf || x == revHalf / 10
}

File: 0009-palindrome-number/0009-palindrome-number.swift
class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        if x < 0 {
            return false
        }

        var temp = x
        var rev = 0

        while temp > 0 {
            rev = rev * 10 + (temp % 10)
            temp /= 10
        }

        return x == rev
    }
}